autosubmit/platforms/platform.py

```diff
@@ -659,15 +637,12 @@ def get_completed_files(self, job_name, retries=0, recovery=False, wrapper_faile
         """
         if recovery:
             retries = 5
-            for i in range(retries):
+            for _ in range(retries):
                 if self.get_file('{0}_COMPLETED'.format(job_name), False, ignore_log=recovery):
                     return True
             return False
-        if self.check_file_exists('{0}_COMPLETED'.format(job_name), wrapper_failed=wrapper_failed):
-            if self.get_file('{0}_COMPLETED'.format(job_name), True, wrapper_failed=wrapper_failed):
-                return True
-            else:
-                return False
+        if self.get_file('{0}_COMPLETED'.format(job_name), True, wrapper_failed=wrapper_failed):
```

Comment:
Hmmm, below I commented that `check_file_exists` must stay. But reading this part of the code :thinking: I would think the same manner. But maybe the code was written this way to prevent unnecessary serialization?

If the platform is a remote one, running in an HPC and the file is really large, then I guess it'd make sense to first check it exists, and then to transfer the file. This, assuming the contract of the function `get_file` does not specify how implementations must handle when the file of `get_file` does not exist.

Probably safer to leave as-was.